### PR TITLE
arm64: refine the fatal handler

### PR DESCRIPTION
**Pull request by lipengfei28**
1. Summary
  The original implement for exception handler is very simple and
haven't framework for breakpoint/watchpoint routine or brk instruction.
  I refine the fatal handler and add framework for debug handler to
register or unregister. this is a prepare for watchpoint/breakpoint implement
2. add arm64_current_el to obtain current EL]

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



## NuttX PR Requirements Review:

**Meets Requirements?:** Partially

**Reasoning:**

* **Summary:** The summary provides a high-level overview but lacks specifics. 
    * **Needs Improvement:**  Clearly state the functional part of the code being changed (e.g., exception handling mechanism) and explicitly describe how the change works (e.g., "introduces a registration mechanism for debug handlers, allowing for breakpoint/watchpoint functionality in the future").

* **Impact:** The "Impact" section is entirely missing.
    * **Needs Addition:** Address all the bullet points in the "Impact" section of the requirements. Pay particular attention to:
       * **New/changed features:** Describe the framework for debug handlers.
       * **Impact on users:** Will users need to modify their code or configuration? 
       * **Impact on hardware/boards:** Are there specific architectures or boards affected?
       * **Documentation:** Does this change necessitate documentation updates? If so, are they included in the PR? 

* **Testing:** The "Testing" section is incomplete.
    * **Needs Improvement:**
        * Specify the build host OS, CPU, compiler, and versions used.
        * List the target architectures, boards, and configurations tested.
        * Provide actual testing logs (or links to them) before and after the change. 

**Recommendations:**

1. **Expand the Summary:** Provide more detailed information about the changes made.
2. **Complete the Impact Section:** Address each bullet point with specific details relevant to your changes.
3. **Fill out the Testing Section:**  Include concrete information about your testing environment and results. 

By addressing these points, you'll make your PR clearer, easier to review, and more likely to be merged.

File: arch/arm64/src/common/arm64_vectors.S [hunk 1]
```diff
@@ -445,41 +442,30 @@ irq_context_switch:
...
-SECTION_FUNC(text, arm64_mode32_error)
-    arm64_exception_context_save x0 x1 sp
-
-    mov    x1, sp
-    mov    x0, #K_ERR_CPU_MODE32
-
-    bl    arm64_fatal_error
+GTEXT(arm64_mode32_handler)
+SECTION_FUNC(text, arm64_mode32_handler)
+    mov    x0, sp
```
Comment (pussuw) [inline arch/arm64/src/common/arm64_vectors.S:457]: Align with line below

File: arch/arm64/src/common/arm64_vectors.S [hunk 2]
```diff
@@ -445,41 +442,30 @@ irq_context_switch:
...
 GTEXT(arm64_fiq_handler)
 SECTION_FUNC(text, arm64_fiq_handler)
 #ifndef CONFIG_ARM64_DECODEFIQ
-    arm64_exception_context_save x0 x1 sp
-
-    mov    x1, sp
-    mov    x0, #K_ERR_SPURIOUS_IRQ /* K_ERR_SPURIOUS_IRQ */
 
-    bl    arm64_fatal_error
+    mov    x0, sp
```
Comment (pussuw) [inline arch/arm64/src/common/arm64_vectors.S:467]: Align with line below

File: arch/arm64/src/common/arm64_vectors.S (hunk 1 above)
Comment (lipengfei28) [reply arch/arm64/src/common/arm64_vectors.S]: done

File: arch/arm64/src/common/arm64_vectors.S (hunk 2 above)
Comment (lipengfei28) [reply arch/arm64/src/common/arm64_vectors.S]: done

Comment (lipengfei28) [reply arch/arm64/src/common/arm64_vectors.S]: done

**Review by pussuw — APPROVED** [review]
LGTM!

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**